src/encoded/static/components/fetched.js

```diff
@@ -1,50 +1,42 @@
-'use strict';
-var React = require('react');
-var cloneWithProps = require('react/lib/cloneWithProps');
-var parseAndLogError = require('./mixins').parseAndLogError;
-var globals = require('./globals');
-var ga = require('google-analytics');
-var _ = require('underscore');
-
-
-var Param = module.exports.Param = React.createClass({
-    contextTypes: {
-        fetch: React.PropTypes.func,
-        session: React.PropTypes.object
-    },
-
-    getDefaultProps: function() {
-        return {type: 'json'};
-    },
-
-    getInitialState: function () {
-        return {
+import React from 'react';
+import PropTypes from 'prop-types';
+import _ from 'underscore';
+import globals from './globals';
+
+
+export class Param extends React.Component {
+    constructor() {
+        super();
+        this.state = {
             fetchedRequest: undefined,
         };
-    },
+        this.receive = this.receive.bind(this);
+        this.fetch = this.fetch.bind(this);
+    }
 
-    componentDidMount: function () {
+    componentDidMount() {
         this.fetch(this.props.url);
-    },
-
-    componentWillUnmount: function () {
-        var xhr = this.state.fetchedRequest;
-        if (xhr) {
-            console.log('abort param xhr');
-            xhr.abort();
-        }
-    },
+    }
 
-    componentWillReceiveProps: function (nextProps, nextContext) {
+    componentWillReceiveProps(nextProps, nextContext) {
         if (!this.state.fetchedRequest && nextProps.url === undefined) return;
         if (this.state.fetchedRequest &&
             nextProps.url === this.props.url &&
-            nextContext.session === this.context.session) return;
+            _.isEqual(nextContext.session, this.context.session)) return;
```

Comment:
This problem was what caused a two-week (at least) delay. The existing code compared whether nextContext.session referred to the same object in memory as this.context.session. This is a _very_ poor assumption, and no longer ever true under React 15. _.isEqual compares whether the two objects have the same contents.